defaults/main.yml

```diff
@@ -16,7 +19,9 @@ cadvisor_port: 9280
 cadvisor_url: https://github.com/google/cadvisor/releases/download/
 
 # Download url
-cadvisor_download: "{{ cadvisor_url }}/{{ cadvisor_version }}/cadvisor"
+cadvisor_download_no_arch: "{{ cadvisor_url }}/{{ cadvisor_version }}/cadvisor"
+cadvisor_download_multi_arch: "{{ cadvisor_url }}/{{ cadvisor_version }}/cadvisor-{{ cadvisor_version }}-{{ cadvisor_cpu_arch }}"
+cadvisor_download: "{{ lookup('vars', 'cadvisor_download_no_arch') if (cadvisor_cpu_arch | length == 0) else lookup('vars','cadvisor_download_multi_arch') }}"
```

Comment:
Since [the 0.45 release of cadvisor](https://github.com/google/cadvisor/releases/tag/v0.45.0) effectively made a breaking change to the name of the assets, I see three possible solutions:

1. also break compatibility in this role and only support `cadvisor 0.45.0` moving forward
2. support binary backwards compatibility using the approach in this PR i.e. 2 internal variables for <0.45 and >=0.45 versions and a variable look-up to decide on the binary layout
3. support backwards compatibility using a single `cadvisor_binary_suffix` variable that would be appended to `cadvisor_download`. Old version would have it empty and e.g. new versions would have `-v0.45.0-linux-amd64`

From our side, we have a limited usage of the `ome.cadvisor` and existing deployments can keep using the current role version. So we do not really have a strong driver/opinion on either option.

Reply:
Option 3 is much cleaner than what I did in option 2. I will update this PR with those changes and leave the version bump in place for now. I think a comment on the `cadvisor_binary_suffix` variable will be necessary either way. If the version bump remains, then `# Empty string "" for all versions < v0.45.0` will suffice. If the decision is to keep `v0.32.0` then a comment explaining the format of `cadvisor_binary_suffix` would be most helpful.

```
# CPU Architecture
# Version >= v0.45.0 use format "-{{ cadvisor_version }}-linux-amd64"
cadvisor_binary_suffix: "-{{ cadvisor_version }}-linux-amd64"
```

Reply:
The formatting issues should be resolved with the latest commit as well.